context/src/main/java/io/opentelemetry/context/DefaultContext.java

```diff
@@ -61,22 +61,22 @@ private DefaultContext(PersistentHashArrayMappedTrie.Node<ContextKey<?>, Object>
   }
 
   @Override
-  public <V> Context with(ContextKey<V> k1, V v1) {
+  public <V> DefaultContext with(ContextKey<V> k1, V v1) {
```

Comment:
Don't think it hurts but is there something we get with covariant types here?

Reply:
Ah realized it does hurt a bit since bytecode instrumentation doesn't work anymore (root() being the important one)

Reply:
You approved this PR. Is this ok, @anuraaga ?

Reply:
Can you explain what is the problem?

Reply:
Yeah - root() is intercepted by auto instrumentation to return an agent managed context. We need the return value to be the public type.

Reply:
No more changes.